R/knitr.R

```diff
@@ -0,0 +1,54 @@
+#' Register the knitr engine for Stan
+#'
+#' Registers CmdStanR's knitr engine [eng_stan()] for processing Stan chunks.
+#'
+#' @param override Override knitr's built-in, RStan-based engine for `stan`.
+#'   See below for details.
+#' @details
+#' If `override = TRUE` (default), this registers CmdStanR's knitr engine as the
+#' engine for `stan` chunks, replacing knitr's built-in, RStan-based engine. If
+#' `override = FALSE`, this registers a `cmdstan` engine so that both engines
+#' may be used in the same R Markdown document. If there is syntax highlighting,
+#' the `cmdstan` chunks will have `stan` syntax highlighting applied to them.
+#' @references
+#' - [Register a custom language engine](https://bookdown.org/yihui/rmarkdown-cookbook/custom-engine.html)
+#' - [Stan language engine](https://bookdown.org/yihui/rmarkdown/language-engines.html#stan)
+#' @export
+register_knitr_engine <- function(override = TRUE) {
+  if (requireNamespace("knitr", quietly = TRUE)) {
```

Comment:
I think we should also have 
```r
else {
  stop("Please install the knitr package.")
}
```
otherwise this will just fail silently if knitr isn't installed, right? 

Or maybe instead of that just

```r
if (!requireNamespace("knitr", quietly = TRUE)) {
  stop("Please install the knitr package.")
}
```
at the top of the function and then there's no need to have rest of the body of the function inside an `if` clause?

Reply:
Good call! I went with the second suggestion.